src/presign/round_one.rs

```diff
@@ -46,56 +48,54 @@ impl Debug for Private {
     }
 }
 
+/// Public information produced in round one of the presign protocol.
 #[derive(Debug, Serialize, Deserialize)]
 pub(crate) struct Public {
-    pub proof: PiEncProof,
+    proof: PiEncProof,
 }
 
+impl Public {
+    pub(crate) fn new(proof: PiEncProof) -> Self {
+        Self { proof }
+    }
+}
```

Comment:
suggestion: This could also be a `From` since we don't expect to ever do anything more complicated here.